### PR TITLE
ci: add buf_breaking_test - missing against field input

### PR DESCRIPTION
**Pull request by mucles**
added `buf_breaking_test`.   it checks the schema in .proto files for breaking changes to maintain backward compatibility.

see [Buf Breaking Bazel Setup Doc](https://buf.build/docs/build-systems/bazel/#example-2)
## Test plan
ci main-dry run

### CODE REVIEW

File: internal/BUILD.bazel [hunk 1]
```diff
@@ -1 +1,20 @@
...
+    against = ":image.binpb",
+    config = "//internal:buf.yaml",
+    targets = [
+        "//internal/own/codeowners/v1:v1_proto",
+        "//internal/repoupdater/v1:v1_proto",
+        "//internal/searcher/v1:v1_proto",
+        "//internal/symbols/v1:v1_proto",
+        "//internal/gitserver/v1:v1_proto",
+        "//internal/grpc/testprotos/news/v1:news_proto",
+    ],  # The Protobuf library
```
Comment (jhchabran) [inline internal/BUILD.bazel:17]:
```suggestion
    ], 
```

File: internal/grpc/testprotos/news/v1/BUILD.bazel [hunk 2]
```diff
@@ -5,15 +5,16 @@ load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 proto_library(
     name = "news_proto",
     srcs = ["news.proto"],
-    visibility = ["//:__subpackages__"],
+    strip_import_prefix = "/internal",  # keep
```
Comment (jhchabran) [inline internal/grpc/testprotos/news/v1/BUILD.bazel:8]: Ah, nice that you found out what was causing a problem here. That being said, didn't you say earlier that this proto file was not useful and is just a left-over from testing things? 

Can't we just drop it?

File: internal/BUILD.bazel [hunk 3]
```diff
@@ -1 +1,20 @@
+load("@rules_buf//buf:defs.bzl", "buf_breaking_test")
+
 exports_files(["buf.yaml"])
+
+buf_breaking_test(
+    name = "v1_proto_breaking",
+    timeout = "short",
+    against = ":image.binpb",
```
Comment (jhchabran) [inline internal/BUILD.bazel:8]: What make you decide to commit this? It's a pretty small file and it's quite convenient to have it. 

What motivated your decision in the end? 

Also, you probably want to document (I think a comment here is fine) how to re-create this file when it needs to be updated.

**Review by jhchabran — CHANGES_REQUESTED** [review]
all good overall, just want to understand the decision around committing the file + we really need some doc / comment to explain how to generate that file again.

**Comment by ggilmore** [comment]
After @mucles and I talked, I think we need the following:

- A way to automatically generate a [buf image ](https://buf.build/docs/reference/images) for every commit to main. 
- A way in CI for buf breaking to automatically to target the [image](https://buf.build/docs/reference/images) in the base branch to check against. 

Right now, the image is simply committed to the repository and there isn't a workflow for updating it. We need a moving target (commit on the base branch for this to be useful). We also need a workflow that ensures that it's only updated on the base branch (updating based off of the latest commit would effecitvely mean that the buf breaking check is only checking the current commit - which isn't useful). 

As such, I think it's best to park this PR until we have more time to commit to solving the above issues. The plan is to make gRPC the default in the 5.2.x release - and remove all the rest code for the 5.3.x. relase. I think it's fine to wait until after 5.2.x to work on this PR, since I think a PR like this naturally falls under the  "making gRPC the only way to do  things in the 5.2.x - 5.3.x interim period" umbrella.

**Pull request closed without merging**